src/components/dateTimePicker/index.js

```diff
@@ -115,20 +115,35 @@ class DateTimePicker extends BaseComponent {
   };
 
   toggleExpandableOverlay = callback => {
-    this.setState({showExpandableOverlay: !this.state.showExpandableOverlay},
-      () => {
-        if (_.isFunction(callback)) {
-          callback();
-        }
-      });
+    this.setState({showExpandableOverlay: !this.state.showExpandableOverlay}, () => {
+      if (_.isFunction(callback)) {
+        callback();
+      }
+    });
   };
 
   onDonePressed = () =>
     this.toggleExpandableOverlay(() => {
       _.invoke(this.props, 'onChange', this.chosenDate);
-      this.setState({chosenDate: this.chosenDate});
+      this.setState({value: this.chosenDate});
     });
 
+  getValue = () => {
```

Comment:
value is a Date object, I would call this method 'getStringValue()'.

Reply:
done